src/http2_connection.erl

```diff
@@ -683,10 +668,9 @@ route_frame(
       stream_id=StreamId,
       type=?RST_STREAM
       },
-   #rst_stream{
-      error_code=EC
-      }},
+   Payload},
   #connection{} = Conn) ->
+    EC = http2_frame_rst_stream:error_code(Payload),
```

Comment:
I don't really get why this is better? it seems like the syntax is a lot clunkier